src/app/src/main/kotlin/org/imperial/mrc/hint/Extensions.kt

```diff
@@ -75,25 +76,36 @@ fun Response.asResponseEntity(): ResponseEntity<String>
         else
         {
             // this is an ADR response, so convert to our response schema
-            formatADRResponse(json)
+            formatADRResponse(json, logger)
         }
 
     }
     catch (e: IOException)
     {
-        logger.error(e.message)
-        ErrorDetail(HttpStatus.INTERNAL_SERVER_ERROR, "Could not parse response.")
-                .toResponseEntity<String>()
+        if (this.body().asString(null).contains("504 Gateway Time-out"))
+        {
+            //Special case of ADR Gateway Timeouts returning HTML responses which cannot be parsed as JSON
+            val message = "ADR request timed out"
+            logger.error(message)
+            ErrorDetail(HttpStatus.GATEWAY_TIMEOUT, message)
+                    .toResponseEntity<String>()
```

Comment:
Fortunately we don't need `<String>` any more, but no need to change - it's consistent with the existing code.